pkg/quarterdeck/db/models/orgs.go

```diff
@@ -81,6 +83,136 @@ func GetOrg(ctx context.Context, id any) (org *Organization, err error) {
 	return org, nil
 }
 
+const (
+	getOrgsForUserSQL = "SELECT id, name, domain, created, modified FROM organizations WHERE id IN (SELECT organization_id FROM organization_users"
+)
+
+// ListOrganizations returns a paginated collection of organizations filtered by the userID.
+// The orgID must be a valid non-zero value of type ulid.ULID,
+// a string representation of a type ulid.ULID, or a slice of bytes
+// The number of organizations resturned is controlled by the prevPage cursor.
+// To return the first page with a default number of results pass nil for the prevPage;
+// Otherwise pass an empty page with the specified PageSize.
+// If the prevPage contains an EndIndex then the next page is returned.
+//
+// A organizations slice with the maximum length of the page size will be returned or an
+// empty (nil) slice if there are no results. If there is a next page of results, e.g.
+// there is another row after the page returned, then a cursor will be returned to
+// compute the next page token with.
+func ListOrgs(ctx context.Context, userID any, prevPage *pagination.Cursor) (organizations []*Organization, cursor *pagination.Cursor, err error) {
+	var user ulid.ULID
+	if user, err = ulids.Parse(userID); err != nil {
+		return nil, nil, err
+	}
+
+	if ulids.IsZero(user) {
+		return nil, nil, ErrMissingModelID
+	}
+
+	if prevPage == nil {
+		// Create a default cursor, e.g. the previous page was nothing
+		prevPage = pagination.New("", "", 0)
+	}
+
+	if prevPage.PageSize <= 0 {
+		return nil, nil, invalid(ErrMissingPageSize)
+	}
+
+	var tx *sql.Tx
+	if tx, err = db.BeginTx(ctx, &sql.TxOptions{ReadOnly: true}); err != nil {
+		return nil, nil, err
+	}
+	defer tx.Rollback()
+
+	// Build parameterized query with WHERE clause
+	// ---------------------------------------------------------------------------------------------------
+	// Query construction with pageSize only:
+	// SELECT id, name, domain, created, modified FROM organizations
+	// WHERE id IN (SELECT organization_id FROM organization_users WHERE user_id=:userID) LIMIT :pageSize
+	// ---------------------------------------------------------------------------------------------------
+	// Query construction with pageSize and endIndex:
+	// SELECT id, name, domain, created, modified FROM organizations
+	// WHERE id IN (SELECT organization_id FROM organization_users WHERE user_id=:userID) AND id > :endIndex LIMIT :pageSize
```

Comment:
Something else we might want to consider is returning a default order for the organizations using an `ORDER BY` clause, however this will probably require a discussion with the frontend team once they get to implementing the multi-org feature.